openquake/commands/plot_assets.py

```diff
@@ -98,7 +98,21 @@ def main(calc_id: int = -1, site_model=False, shapefile_path=None,
     w, h = maxx - minx, maxy - miny
     ax.set_xlim(minx - 0.2 * w, maxx + 0.2 * w)
     ax.set_ylim(miny - 0.2 * h, maxy + 0.2 * h)
-    ax.legend()
+
+    try:
+        iso_codes = [
+            iso_code.decode('utf8')
+            for iso_code in
+            dstore['assetcol/tagcol/ID_0'][numpy.unique(assetcol['ID_0'])]]
```

Comment:
For performance reasons, the right way to do this is as follows:
```python
ALL_ID_0 =  dstore['assetcol/tagcol/ID_0'][:]
ID_0 = ALL_ID_0[numpy.unique(assetcol['ID_0'])]
```

Reply:
Done